typingturtle-keyboard.patch

```diff
@@ -0,0 +1,26 @@
+diff --git a/keyboard.py b/keyboard.py
+index 621f71c..aaec3d1 100644
+--- a/keyboard.py
++++ b/keyboard.py
+@@ -104,14 +104,14 @@ def _is_olpcm_model():
+     keyboard, and 'olpc' for membrane keyboard.
+ 
+     """
+-    code = None
+-    p = subprocess.Popen(["setxkbmap", "-query"], stdout=subprocess.PIPE)
+-    out, err = p.communicate()
++    # code = None
++    # p = subprocess.Popen(["setxkbmap", "-query"], stdout=subprocess.PIPE)
++    # out, err = p.communicate()
+     
+-    for line in out.decode('utf-8').splitlines():
+-        if line.startswith('model:'):
+-            code = line.split()[1]
+-    return code == 'olpcm'
```

Comment:
What's the rationale behind this patch?

Reply:
```File "/app/share/sugar/activities/TypingTurtle.activity/mainscreen.py", line 264, in lesson_clicked_cb
    self.activity.push_screen(lessonscreen.LessonScreen(self.visible_lesson, self.keyboard_images, self.activity))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/share/sugar/activities/TypingTurtle.activity/lessonscreen.py", line 141, in __init__
    self.keyboard.set_layout(keyboard.get_layout())
                             ^^^^^^^^^^^^^^^^^^^^^
  File "/app/share/sugar/activities/TypingTurtle.activity/keyboard.py", line 119, in get_layout
    if _is_olpcm_model():
       ^^^^^^^^^^^^^^^^^
  File "/app/share/sugar/activities/TypingTurtle.activity/keyboard.py", line 110, in _is_olpcm_model
    p = subprocess.Popen(["setxkbmap", "-query"], stdout=subprocess.PIPE)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.11/subprocess.py", line 1955, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'setxkbmap'
```
This error was generated when we press on `start lesson`  after porting to flatpak.
In order to solve this I had to remove below mentioned line
```
    p = subprocess.Popen(["setxkbmap", "-query"], stdout=subprocess.PIPE)
    out, err = p.communicate()
```
After removing this the `for loop` became invalid therefore I removed that too.

This is basically used to detect whether The keyboard layout is `olpcm` or `olpc`. Since we are not using OLPC laptops for flatpak I made the `_is_olpcm_model` function to return `None`. 
This will further return `OLPC_LAYOUT` when `get_layout` function is called. This was done because both `olpcm` and `olpc` layout are not upto the mark and we are trying to make the changes in `olpc` layout in upstream.

Reply:
Have you tried adding setxkbmap to the flatpak ? e.g., https://github.com/search?q=org%3Aflathub%20setxkbmap&type=code